src/Profiles.ts

```diff
@@ -1091,6 +1126,70 @@ export class Profiles {
         return filteredProfile;
     }
 
+    public async getCombinedProfile(serviceProfile: IProfileLoaded, baseProfile: IProfileLoaded) {
+
+        // TODO: This needs to be improved
+        // The idea is to handle all type of ZE Profiles
+
+        // This check will handle service profiles that hase username and password
+        if (serviceProfile.profile.user && serviceProfile.profile.password) {
+            return serviceProfile;
+        }
+
+        // This check is for optional credentials
+        if (baseProfile && serviceProfile.profile.host && serviceProfile.profile.port &&
+            ((baseProfile.profile.host !== serviceProfile.profile.host) && baseProfile.profile.port !== serviceProfile.profile.port)) {
+                return serviceProfile;
+        }
+
+        // This process combines the information from baseprofile to serviceprofile and create a new session
+        const cmdArgs: ICommandArguments = {
+            $0: "zowe",
+            _: [""],
+            host: serviceProfile.profile.host ? serviceProfile.profile.host :
+            (baseProfile ? baseProfile.profile.host : undefined),
+            port: serviceProfile.profile.port ? serviceProfile.profile.port :
+            (baseProfile ? baseProfile.profile.port : 0),
+            basePath: serviceProfile.profile.basePath ? serviceProfile.profile.basePath :
+            (baseProfile ? baseProfile.profile.basePath : undefined),
+            rejectUnauthorized: serviceProfile.profile.rejectUnauthorized !== null ?
+            serviceProfile.profile.rejectUnauthorized :
+            (baseProfile ? baseProfile.profile.rejectUnauthorized : true),
+            user: serviceProfile.profile.user ? serviceProfile.profile.user :
+            (baseProfile ? baseProfile.profile.user : undefined),
+            password: serviceProfile.profile.password ? serviceProfile.profile.password :
+            (baseProfile ? baseProfile.profile.password : undefined),
+            tokenType: baseProfile.profile.tokenType,
+            tokenValue: (baseProfile && !serviceProfile.profile.password) ? baseProfile.profile.tokenValue : undefined
+        };
+
+        const combinedSession = await zowe.ZosmfSession.createBasicZosmfSessionFromArguments(cmdArgs);
+
+        // For easier debugging, move serviceProfile to updatedServiceProfile and then update it with combinedSession
+        const updatedServiceProfile: IProfileLoaded = serviceProfile;
+        updatedServiceProfile.profile.host = combinedSession.ISession.hostname;
+        updatedServiceProfile.profile.port = combinedSession.ISession.port;
+        updatedServiceProfile.profile.basePath = combinedSession.ISession.basePath;
+        updatedServiceProfile.profile.rejectUnauthorized = combinedSession.ISession.rejectUnauthorized;
+        updatedServiceProfile.profile.tokenType = combinedSession.ISession.tokenType;
+        updatedServiceProfile.profile.tokenValue = combinedSession.ISession.tokenValue;
```

Comment:
This code also needs to use the schema as unless these are all mandatory fields for profiles? Not sure if every profile type needs to specify `rejectUnauthorized`? Have to look that up.

Reply:
Hello, I have tried to address this [here](https://github.com/zowe/vscode-extension-for-zowe/blob/49e340be9076aab83af056aa9594a537f274378b/src/Profiles.ts#L1180)...